src/App.tsx

```diff
@@ -12,74 +23,302 @@ function App() {
   const [nodeOptions, setNodeOptions] = useState<NodeOption[]>([
     { NodeName: 'All', ApiURL: 'allNodes' },
   ]);
+
+  const [alertDismissed, setAlertDismissed] = useState<boolean>(false);
+
+  const [loading, setLoading] = useState<boolean>(false);
+
   const [result, setResult] = useState<Result[] | null>(null);
 
+  const [node, setNode] = useState<FieldInput>([{ label: 'All', id: 'allNodes' }]);
+  const [minAge, setMinAge] = useState<number | null>(null);
+  const [maxAge, setMaxAge] = useState<number | null>(null);
+  const [sex, setSex] = useState<FieldInput>(null);
+  const [diagnosis, setDiagnosis] = useState<FieldInput>(null);
+  const [isControl, setIsControl] = useState<boolean>(false);
+  const [minNumSessions, setMinNumSessions] = useState<number | null>(null);
+  const [assessmentTool, setAssessmentTool] = useState<FieldInput>(null);
+  const [imagingModality, setImagingModality] = useState<FieldInput>(null);
+  const [searchParams, setSearchParams] = useSearchParams();
+
   useEffect(() => {
-    async function fetchOptions(
-      dataElementURI: string,
-      setOptions: (options: AttributeOption[]) => void
-    ) {
+    async function tryThisOptionsGetter(dataElementURI: string) {
```

Comment:
as much as I like the function name, maybe we should refactor this to something more typical. Maybe `getAttributes()`?

Reply:
oh yea forgot about that.